controllers/mentor.go

```diff
@@ -0,0 +1,85 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"kwoc-backend/middleware"
+	"kwoc-backend/utils"
+	"net/http"
+
+	"kwoc-backend/models"
+
+	"gorm.io/gorm"
+)
+
+type RegisterMentorReqFields struct {
+	Username string `json:"username"`
+	Name     string `json:"name"`
+	Email    string `json:"email"`
+}
+
```

Comment:
Shall we also take work/organisation/company as a parameter this year? Would be nice to know and show that the mentor is a student too or works somewhere.

Reply:
We can do that but let's make a separate issue to track new features.